File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int cnt =0;
        int ans = 0;
        for(int num :  nums){
            if(cnt==0){
                ans = num;
            }
            if(num == ans){
                cnt ++;
            }
            else{
                cnt--;
            }

        }
        return ans;
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:

int majorityElement(vector<int>& nums) {
    int n = nums.size();
    int cnt = 0;
    int ele = 0;

    for(int i = 0; i < n; i++) {
        if (cnt == 0) {
            ele = nums[i];
            cnt = 1;
        } else if (nums[i] == ele) {
            cnt++;
        } else {
            cnt--;
        }
    }

    int count = 0;
    for(int i = 0; i < n; i++) {
        if (nums[i] == ele) {
            count++;
        }
    }
    if (count > n / 2) {
        return ele;
    }
    
    return -1; 
}

};